guvnor-ala/guvnor-ala-wildfly-provider/src/main/java/org/guvnor/ala/wildfly/access/WildflyClient.java

```diff
@@ -0,0 +1,296 @@
+                .build();
+
+        final HttpPost post = new HttpPost( "http://" + host + ":" + managementPort + "/management" );
+
+        post.addHeader( "X-Management-Client-Name", "GUVNOR-ALA" );
+
+        // the DMR operation
+        ModelNode operation = new ModelNode();
+        operation.get( "operation" ).set( "deploy" );
+        operation.get( "address" ).add( "deployment", deploymentName );
+
+        try {
+            post.setEntity( new StringEntity( operation.toJSONString( true ), APPLICATION_JSON ) );
+        } catch ( Exception e ) {
+            e.printStackTrace();
+        }
+
+        try {
+            HttpResponse response = httpclient.execute( post );
+            return response.getStatusLine().getStatusCode();
+        } catch ( IOException ex ) {
+            ex.printStackTrace();
+            getLogger( WildflyClient.class.getName() ).log( SEVERE, null, ex );
+        }
+        return -1;
+    }
+
+    public int stop( String deploymentName ) {
+        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+        credsProvider.setCredentials(
+                new AuthScope( host, managementPort ),
+                new UsernamePasswordCredentials( user, password ) );
+
+        CloseableHttpClient httpclient = custom()
+                .setDefaultCredentialsProvider( credsProvider )
+                .build();
+
+        final HttpPost post = new HttpPost( "http://" + host + ":" + managementPort + "/management" );
+
+        post.addHeader( "X-Management-Client-Name", "GUVNOR-ALA" );
+
+        // the DMR operation
+        ModelNode operation = new ModelNode();
+        operation.get( "operation" ).set( "undeploy" );
+        operation.get( "address" ).add( "deployment", deploymentName );
+
+        try {
+            post.setEntity( new StringEntity( operation.toJSONString( true ), APPLICATION_JSON ) );
+        } catch ( Exception e ) {
+            e.printStackTrace();
+        }
+
+        try {
+            HttpResponse response = httpclient.execute( post );
+            return response.getStatusLine().getStatusCode();
+        } catch ( IOException ex ) {
+            ex.printStackTrace();
+            getLogger( WildflyClient.class.getName() ).log( SEVERE, null, ex );
+        }
+        return -1;
+    }
+
+    public void restart( String id ) {
+
+    }
+
+    public WildflyAppState getAppState( String deploymentName ) {
+        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+        credsProvider.setCredentials(
+                new AuthScope( host, managementPort ),
+                new UsernamePasswordCredentials( user, password ) );
+
+        CloseableHttpClient httpclient = custom()
+                .setDefaultCredentialsProvider( credsProvider )
+                .build();
+
+        final HttpPost post = new HttpPost( "http://" + host + ":" + managementPort + "/management" );
+
+        post.addHeader( "X-Management-Client-Name", "GUVNOR-ALA" );
```

Comment:
HAL is only used for uploads as far as I know for the other commands we can use GUVNOR-ALA